internal/controller/vmprometheusconverter_controller.go

```diff
@@ -151,46 +161,86 @@ func NewConverterController(ctx context.Context, baseClient *kubernetes.Clientse
 	}); err != nil {
 		return nil, fmt.Errorf("cannot add service_monitor handler: %w", err)
 	}
-	c.amConfigInf = cache.NewSharedIndexInformer(
+
+	// AlertmanagerConfig/v1alpha1
+	amConfigInf := cache.NewSharedIndexInformer(
 		&cache.ListWatch{
 			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
-				var objects v1alpha1.AlertmanagerConfigList
-				if err := k8stools.ListObjectsByNamespace(ctx, rclient, config.MustGetWatchNamespaces(), func(dst *v1alpha1.AlertmanagerConfigList) {
+				var objects promv1alpha1.AlertmanagerConfigList
+				if err := k8stools.ListObjectsByNamespace(ctx, rclient, config.MustGetWatchNamespaces(), func(dst *promv1alpha1.AlertmanagerConfigList) {
 					objects.Items = append(objects.Items, dst.Items...)
 				}); err != nil {
 					return nil, fmt.Errorf("cannot list alertmanager_configs: %w", err)
 				}
 				return &objects, nil
 			},
 			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
-				return k8stools.NewObjectWatcherForNamespaces[v1alpha1.AlertmanagerConfigList](ctx, rclient, "alertmanager_configs", config.MustGetWatchNamespaces())
+				return k8stools.NewObjectWatcherForNamespaces[promv1alpha1.AlertmanagerConfigList](ctx, rclient, "alertmanager_configs", config.MustGetWatchNamespaces())
 			},
 		},
-		&v1alpha1.AlertmanagerConfig{},
+		&promv1alpha1.AlertmanagerConfig{},
 		resyncPeriod,
-		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
-	if _, err := c.amConfigInf.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+	)
+	if _, err := amConfigInf.AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc:    c.CreateAlertmanagerConfig,
 		UpdateFunc: c.UpdateAlertmanagerConfig,
 	}); err != nil {
 		return nil, fmt.Errorf("cannot add alertmanager_config handler: %w", err)
 	}
+	c.amConfigInfs = append(c.amConfigInfs, informer{
+		informer: amConfigInf,
+		version:  promv1alpha1.SchemeGroupVersion.String(),
+		kind:     promv1alpha1.AlertmanagerConfigKind,
+	})
+
+	// AlertmanagerConfig/v1beta1
```

Comment:
Can we check if cluster has installed prometheus v1beta1 CRD first, and skip watching if there is not?
Now running with env that doesn't install v1beta1.AlertmanagerConfig gives errors
```
{"level":"info","ts":"2024-06-20T17:22:14+08:00","msg":"pkg/mod/k8s.io/client-go@v0.30.2/tools/cache/reflector.go:232: failed to list *v1beta1.AlertmanagerConfig: cannot list alertmanager_configs: cannot list objects at cluster scope: no kind is registered for the type v1beta1.AlertmanagerConfigList in scheme \"pkg/runtime/scheme.go:100\""}
{"level":"error","ts":"2024-06-20T17:22:14+08:00","msg":"pkg/mod/k8s.io/client-go@v0.30.2/tools/cache/reflector.go:232: Failed to watch *v1beta1.AlertmanagerConfig: failed to list *v1beta1.AlertmanagerConfig: cannot list alertmanager_configs: cannot list objects at cluster scope: no kind is registered for the type v1beta1.AlertmanagerConfigList in scheme \"pkg/runtime/scheme.go:100\"","stacktrace":"k8s.io/klog/v2.(*loggingT).output\n\t/Users/haley/go/pkg/mod/k8s.io/klog/v2@v2.120.1/klog.go:912\nk8s.io/klog/v2.(*loggingT).printWithInfos\n\t/Users/haley/go/pkg/mod/k8s.io/klog/v2@v2.120.1/klog.go:740\nk8s.io/klog/v2.(*loggingT).printDepth\n\t/Users/haley/go/pkg/mod/k8s.io/klog/v2@v2.120.1/klog.go:722\nk8s.io/klog/v2.ErrorDepth\n\t/Users/haley/go/pkg/mod/k8s.io/klog/v2@v2.120.1/klog.go:1591\nk8s.io/apimachinery/pkg/util/runtime.logError\n\t/Users/haley/go/pkg/mod/k8s.io/apimachinery@v0.30.2/pkg/util/runtime/runtime.go:115\nk8s.io/apimachinery/pkg/util/runtime.HandleError\n\t/Users/haley/go/pkg/mod/k8s.io/apimachinery@v0.30.2/pkg/util/runtime/runtime.go:109\nk8s.io/client-go/tools/cache.DefaultWatchErrorHandler\n\t/Users/haley/go/pkg/mod/k8s.io/client-go@v0.30.2/tools/cache/reflector.go:150\nk8s.io/client-go/tools/cache.(*Reflector).Run.func1\n\t/Users/haley/go/pkg/mod/k8s.io/client-go@v0.30.2/tools/cache/reflector.go:299\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/Users/haley/go/pkg/mod/k8s.io/apimachinery@v0.30.2/pkg/util/wait/backoff.go:226\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/Users/haley/go/pkg/mod/k8s.io/apimachinery@v0.30.2/pkg/util/wait/backoff.go:227\nk8s.io/client-go/tools/cache.(*Reflector).Run\n\t/Users/haley/go/pkg/mod/k8s.io/client-go@v0.30.2/tools/cache/reflector.go:297\nk8s.io/apimachinery/pkg/util/wait.(*Group).StartWithChannel.func1\n\t/Users/haley/go/pkg/mod/k8s.io/apimachinery@v0.30.2/pkg/util/wait/wait.go:55\nk8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1\n\t/Users/haley/go/pkg/mod/k8s.io/apimachinery@v0.30.2/pkg/util/wait/wait.go:72"}
{"level":"info","ts":"2024-06-20T17:23:03+08:00","msg":"pkg/mod/k8s.io/client-go@v0.30.2/tools/cache/reflector.go:232: failed to list *v1beta1.AlertmanagerConfig: cannot list alertmanager_configs: cannot list objects at cluster scope: no kind is registered for the type v1beta1.AlertmanagerConfigList in scheme \"pkg/runtime/scheme.go:100\""}
{"level":"error","ts":"2024-06-20T17:23:03+08:00","msg":"pkg/mod/k8s.io/client-go@v0.30.2/tools/cache/reflector.go:232: Failed to watch *v1beta1.AlertmanagerConfig: failed to list *v1beta1.AlertmanagerConfig: cannot list alertmanager_configs: cannot list objects at cluster scope: no kind is registered for the type v1beta1.AlertmanagerConfigList in scheme \"pkg/runtime/scheme.go:100\"","stacktrace":"k8s.io/klog/v2.(*loggingT).output\n\t/Users/haley/go/pkg/mod/k8s.io/klog/v2@v2.120.1/klog.go:912\nk8s.io/klog/v2.(*loggingT).printWithInfos\n\t/Users/haley/go/pkg/mod/k8s.io/klog/v2@v2.120.1/klog.go:740\nk8s.io/klog/v2.(*loggingT).printDepth\n\t/Users/haley/go/pkg/mod/k8s.io/klog/v2@v2.120.1/klog.go:722\nk8s.io/klog/v2.ErrorDepth\n\t/Users/haley/go/pkg/mod/k8s.io/klog/v2@v2.120.1/klog.go:1591\nk8s.io/apimachinery/pkg/util/runtime.logError\n\t/Users/haley/go/pkg/mod/k8s.io/apimachinery@v0.30.2/pkg/util/runtime/runtime.go:115\nk8s.io/apimachinery/pkg/util/runtime.HandleError\n\t/Users/haley/go/pkg/mod/k8s.io/apimachinery@v0.30.2/pkg/util/runtime/runtime.go:109\nk8s.io/client-go/tools/cache.DefaultWatchErrorHandler\n\t/Users/haley/go/pkg/mod/k8s.io/client-go@v0.30.2/tools/cache/reflector.go:150\nk8s.io/client-go/tools/cache.(*Reflector).Run.func1\n\t/Users/haley/go/pkg/mod/k8s.io/client-go@v0.30.2/tools/cache/reflector.go:299\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/Users/haley/go/pkg/mod/k8s.io/apimachinery@v0.30.2/pkg/util/wait/backoff.go:226\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/Users/haley/go/pkg/mod/k8s.io/apimachinery@v0.30.2/pkg/util/wait/backoff.go:227\nk8s.io/client-go/tools/cache.(*Reflector).Run\n\t/Users/haley/go/pkg/mod/k8s.io/client-go@v0.30.2/tools/cache/reflector.go:297\nk8s.io/apimachinery/pkg/util/wait.(*Group).StartWithChannel.func1\n\t/Users/haley/go/pkg/mod/k8s.io/apimachinery@v0.30.2/pkg/util/wait/wait.go:55\nk8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1\n\t/Users/haley/go/pkg/mod/k8s.io/apimachinery@v0.30.2/pkg/util/wait/wait.go:72"}
```
And looks like it should be added as a feature in changelog as well.

Reply:
removed v1beta1.AlertmanagerConfig informer, as it's CRDs are not available yet

Reply:
Okay, so prometheus created v1beta1 AlertmanagerConfig but still [not migrate to it](https://github.com/prometheus-community/helm-charts/pull/3938) in prometheus-operator or their chart.
Could you also remove other codes which using `promv1beta1.AlertmanagerConfig`
https://github.com/VictoriaMetrics/operator/blob/232d977c68aff75efa7e7b53d3273fd4896c19b1/internal/controller/vmprometheusconverter_controller.go#L487-L488

Reply:
done